src/base.satyg

```diff
@@ -1,13 +1,20 @@
-@import: base0
-@import: __sub__/base-sub
+@stage: persistent
+
+%% Short-circuit conjunction operator.
+%%
+%% e.g. [if (i < String.length s) &&&- (fun () -> Char.equal (Char.at i s) Char.space) then `yes` else `no`]
+let ( &&&- ) b1 b2f = if b1 then b2f () else false
+
+%% Short-circuit disjunction operator. See also [(&&&-)].
+let ( |||- ) b1 b2f = if b1 then true else b2f ()
```

Comment:
I am totally in favor of adding these kinds of combinatory to our library!

However, currently we have no consensus on how their concrete interfaces should look like. Especially my concern is about their names and types i.e. the first operand is a mere value while the second one is a thunk.
I think we will need some more experiences to identify the best practice before making them our official, stable API.

So, my feeling is that leaving a comment here saying this is not a stable API should be a viable option for the moment. Thoughts?

Reply:
Thank you for quick review!

> my feeling is that leaving a comment here saying this is not a stable API should be a viable option for the moment. Thoughts?

Yes, that sounds a good choice to me, since at least currently `&&&-` and `|||-` are only for internal use and adding them is not the main purpose of this PR but a kind of “side effects”.

I will add such kinds of comment here.

Reply:
Added comments.